blackbox/docs/sql/administration/privileges.txt

```diff
@@ -12,47 +12,84 @@ Cluster Wide Privileges
-issue ``INSERT`` statement to any table and schema in the cluster.
-There is no option to ``GRANT`` or ``REVOKE`` privileges on object
-level, such as table or schema.
+The superuser uses ``GRANT``, ``DENY`` and ``REVOKE`` statements to control
+access to the resource. The privileges are either applied on the whole cluster
+or on instances of objects such as schema, or table.
 
 .. note::
 
     Superusers are allowed to execute any statement without any privilege
     checks. Currently only ``DQL``, ``DML`` and ``DDL`` privileges can be
     granted. Any statements which are not allowed with those privileges,
-    such as ``GRANT`` and ``REVOKE``, can only be issued by a superuser.
+    such as ``GRANT``, ``DENY`` and ``REVOKE``, can only be issued by
+    a superuser.
 
-``Privilege Types``
-====================
+Privilege Types
+===============
 
 ``DQL``
 .......
 
 Granting ``Data Query Language (DQL)`` privilege to a user, indicates that this
 user is allowed to execute ``SELECT``, ``SHOW``, ``REFRESH``, ``COPY TO``,
 ``EXPLAIN`` and ``SET SESSION`` statements, as well as using the available
-user defined functions.
+user defined functions, on the object for which the privilege applies.
 
 ``DML``
 .......
 
 Granting ``Data Manipulation Language (DML)`` privilege to a user, indicates
 that this user is allowed to execute ``INSERT``, ``COPY FROM```, ``UPDATE``
-and ``DELETE`` statements.
+and ``DELETE`` statements, on the object for which the privilege applies.
 
 ``DDL``
 .......
 
 Granting ``Data Definition Language (DDL)`` privilege to a user, indicates that
-this user is allowed to execute ``CREATE``, ``DROP`` and ``ALTER`` statements.
+this user is allowed to execute ``CREATE``, ``DROP`` and ``ALTER`` statements,
+on the object for which the privilege applies.
+
+
+Hierarchical Inheritance of Privileges
+======================================
+.. hide:
+
+    cr> CREATE user riley;
+    CREATE OK, 1 row affected (... sec)
+
+    cr> CREATE user kala;
+    CREATE OK, 1 row affected (... sec)
+
+Privileges can be managed on three different levels, namely: ``CLUSTER``, ``SCHEMA``
+and ``TABLE``.
+
+When a privilege is assigned to an object, the privilege will propagate down the
+object hierarchy. Privileges defined for a child object will always override those
+from a parent object. For example::
+
+    cr> GRANT DQL ON SCHEMA doc TO riley;
+    GRANT OK, 1 row affected (... sec)
+
+This statement will grant ``DQL`` privilege to user riley on all the tables
+and functions of the ``doc`` schema::
+
+    cr> DENY DQL ON TABLE doc.accounting TO riley;
+    GRANT OK, 1 row affected (... sec)
+
+This statement will deny ``DQL`` privilege to user riley on the ``doc`` schema table
+``doc.accounting``. However, user riley, will still have ``DQL`` privilege on all the
+other tables of the ``doc`` schema.
+
+.. note::
+
+    Privileges can be managed on all schemas and tables of the cluster,
+    including the ``sys`` schema.
```

Comment:
Isn't the `sys.users` table still kind of special in that it's super-user only? Or will it be possible to grant users permission to it as well?